File: exercises/D202/solution.py
import math 

n = int(input())
m = int(math.sqrt(n))
for i in range(1, m + 1):
    if n % i == 0:
        print(i)
if m * m == n:
    m -= 1
for i in range(m, 0, -1):
    if n % i == 0:
        print(n // i)

File: exercises/D202/solution.java
import java.util.*;

public class D202 {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int m = (int)Math.sqrt(n);
        for (int i = 1; i <= m; i++) {
            if (n % i == 0) {
                System.out.println(i);
            }
        }
        if (m * m == n) {
            m--;
        }
        for (int i = m; i >= 1; i--) {
            if (n % i == 0) {
                System.out.println(n / i);
            }
        }
    }
}
